kubeflow/core/util.libsonnet

```diff
@@ -31,5 +33,39 @@
       else [],
   }.result,
 
+  // Convert an object's properties into an array of values
+  toList:: [$[key] for key in std.objectFields($)],
+
+  // Convert an array of kubernetes resources into a  map[kind/name, resource]
+  toMap:: function(params) {
```

Comment:
Why is toMap taking params as an argument?

Reply:
I believe I understand. Does the following make sense?

1. Remove params from toMap ([prometheus.libsonnet](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/prometheus.libsonnet) is an example where parts is a map and all is parts)
2. Build the map first, then the parts which reference the map eg: [map.values()]  

Question: What about repeating resources within a libsonnet? For example jupyterhub.libsonnet has 2 Services, 2 ServiceAccounts, 2 Roles, 2 RoleBindings. Should we deal with it only if needed? For example:
```
local jup = {
  Service: {
    "tf-hub-0": {foo:"bar",},
    "tf-hub-lb": {bar:"baz",},
  },
};
```
so the update would look like
```
jup +  {
   Service+: {
    "tf-hub-0"+: {one: "two",},
  },
}
```
yielding
```
{
   "Service": {
      "tf-hub-0": {
         "foo": "bar",
         "one": "two"
      },
      "tf-hub-lb": {
         "bar": "baz"
      }
   }
}
```
for non-repeating resources it would look like your example with StatefulSet or would that be too confusing?

Reply:
I don't think the top level items in the dictionary should be based on type. They should be based on what they do; e.g. if there are multiple services each service should be its own item.

Reply:
so how bout
```
{
  ConfigMap:: 
  HubService::
  HubServiceAccount::
  HubRole::
  HubRoleBinding::
  NotebookService::
  NotebookServiceAccount::
  NotebookRole::
  NotebookRoleBinding::  
  StatefulSet::
}
```

where the map is a set of functional hidden names

Reply:
parts would be in the top level mapping
```
{
  ConfigMap:: 
  HubService::
  HubServiceAccount::
  HubRole::
  HubRoleBinding::
  NotebookService::
  NotebookServiceAccount::
  NotebookRole::
  NotebookRoleBinding::  
  StatefulSet::

  parts:: [ ConfigMap:: HubService:: ... ]
}